File: CarCareHub_/CarCareHub.Model/FirmaAutodijelova.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class FirmaAutodijelova
    {
        public int FirmaAutodijelovaID { get; set; }

        public string? NazivFirme { get; set; }

        public string? Adresa { get; set; }

        public int? GradId { get; set; }
        public string? JIB { get; set; }
        public string? MBS { get; set; }

        public string? Telefon { get; set; }
        public bool Vidljivo { get; set; }
        public string? Email { get; set; }
        public string? Username { get; set; }

        public string? Password { get; set; }
    

        public string? LozinkaSalt { get; set; }
        public string? LozinkaHash { get; set; }

        [Column(TypeName = "VARBINARY(MAX)")]
        public byte[]? SlikaProfila { get; set; }

        public int? UlogaId { get; set; }
        public virtual Grad? Grad { get; set; }
        public virtual Uloge? Uloga { get; set; }
        //public virtual Drzava? Drzava { get; set; }
        //public virtual ICollection<BPAutodijeloviAutoservis> BPAutodijeloviAutoservis { get; set; } = new List<BPAutodijeloviAutoservis>();


        // public int? IzvjestajId { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_mobilna/lib/models/firmaautodijelova.dart
import 'package:flutter_mobile/models/grad.dart';
import 'package:json_annotation/json_annotation.dart';
part 'firmaautodijelova.g.dart';

@JsonSerializable()
class FirmaAutodijelova {
  int firmaAutodijelovaID;
  String? nazivFirme;
  String? adresa;
  int? gradId;
  final Grad? grad;
  String? jib;
  String? mbs;
  String? telefon;
  String? email;
  String? password;
  String? slikaProfila;
  String? username;
  int? ulogaId;
  String? passwordAgain;
bool? vidljivo;
  



  FirmaAutodijelova(
      this.firmaAutodijelovaID,
      this.vidljivo,
      this.nazivFirme,
      this.adresa,
      this.email,
      this.gradId,
      this.jib,
      this.mbs,
      this.password,
      this.slikaProfila,
      this.telefon,
      this.ulogaId,
      this.grad,
      this.passwordAgain,
      this.username);

  factory FirmaAutodijelova.fromJson(Map<String, dynamic> json) =>
      _$FirmaAutodijelovaFromJson(json);

  Map<String, dynamic> toJson() => _$FirmaAutodijelovaToJson(this);
}
